public/javascripts/playlistDropdown.js

```diff
@@ -75,26 +75,71 @@ var playlistDropdownComponent = (function() {
                             ]),
                             m('.modal-body', [
                                 m('label', 'Enter Playlist Name'),
-                                m('input.form-control[name=playlist type=text]', {
+                                m('input#newPlaylistInput.form-control[name=playlist type=text]', {
                                     value: 'New Playlist',
-                                    onchange: m.withAttr('value', args.playlistName),
                                     config: function(elem, isInit) {
                                         if (!isInit) {
                                             enter(elem)
                                         }
                                     }
                                 }),
                                 m('br'),
-                                m('button.btn.btn-success#create', {
+                                m('button.btn.btn-default#create', {
                                     "data-dismiss": "modal",
                                     onclick: function() {
+                                        args.playlistName($('input#newPlaylistInput').val())
                                         addPlaylist(args.playlistName(), args.url)
                                         $('#newPlaylist').modal('hide')
                                     }
                                 }, 'Create')
                             ])
                         ])
                     ])
+                ]),
+                m('#choosePlaylist.modal.fade[role=dialog]', [
+                    m('.modal-dialog.modal-sm', [
+                        m('.modal-content', [
+                            m('.modal-header', [
+                                m('h4', {
+                                    style: {
+                                        'margin-left': '40px'
+                                    }
+                                }, 'Select Playlist', [
```

Comment:
consider pulling this out into a component or function to flatten the code a bit.